eox_nelp/notifications/notify_course_due_date.py

```diff
@@ -0,0 +1,78 @@
+"""Notificaty course date module. This manage the delivery of an
+upcoming_course_due_date
+"""
+import logging
+
+from django.utils import timezone
+
+from eox_nelp.edxapp_wrapper.bulk_email import CourseEmailTemplate, get_course_email_context
+from eox_nelp.edxapp_wrapper.student import CourseEnrollment
+from eox_nelp.notifications.utils import (
+    get_xblock_course_from_usage_key_string,
+    get_xblock_from_usage_key_string,
+    send_email_multialternative,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def notify_upcoming_course_due_date(upcoming_course_due_date):
+    """Send the correspongind email of an UpcomingCourseDate object.
+    This is using the bulk-email template that already exists.
+    Args:
+        upcoming_course_due_date (UpcomingCourseDate): UpcomingCourseDate object to sent emails.
+    """
+    #  pylint: disable=too-many-locals
+    subsection_key_string = str(upcoming_course_due_date.location_id)
+    enrollments_qs = CourseEnrollment.objects.filter(course__id=upcoming_course_due_date.course_id, is_active=True)
+    recipient_emails = [enrollment.user.email for enrollment in enrollments_qs if enrollment.user.email]
+
+    subsection_xblock = get_xblock_from_usage_key_string(subsection_key_string)
+    course_xblock = get_xblock_course_from_usage_key_string(subsection_key_string)
+
+    if (not subsection_xblock) or (not recipient_emails) or (not course_xblock):
```

Comment:
I know I said use `or` but  you mentioned something with `and`  and now I'm thinking in something like this suggestion
```suggestion
    if not (subsection_xblock and recipient_emails and course_xblock):
```